services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -844,10 +845,18 @@ open class HankeServiceImpl(
                 status = HankeStatus.DRAFT,
                 generated = true,
             ),
-            perustaja = generatePerustajaFrom(cableReport.applicationData),
+            perustaja = perustajaFrom(cableReport.applicationData),
         )
 
-    private fun generatePerustajaFrom(cableReport: CableReportApplicationData): Perustaja {
+    private fun limitHankeName(input: String): String {
+        val limit = MAXIMUM_HANKE_NIMI_LENGTH
+        if (input.length > limit) {
+            logger.info { "Hanke name too long, limited to first $limit characters." }
+        }
+        return input.take(limit)
+    }
```

Comment:
Modifying the name is not necessary unless it's over the maximum length. Skipping it is not very important, but since the if is there already, we might as well use it for this as well.

Aliasing the constant only makes it less efficient. Without declaring the value, the function can use an expression body.

Since this function is custom-made to check the hanke name and not a generic limit for string length, the parameter could be named `name`.
```suggestion
    private fun limitHankeName(name: String): String =
        if (name.length > MAXIMUM_HANKE_NIMI_LENGTH) {
            logger.info {
                "Hanke name too long, limited to first $MAXIMUM_HANKE_NIMI_LENGTH characters."
            }
            name.take(MAXIMUM_HANKE_NIMI_LENGTH)
        } else {
            name
        }
```

Reply:
Modified as suggested.